src/index.js

```diff
@@ -0,0 +1,226 @@
+  const readKeyRole = "read";
+  const readKey = apiKeys.find(
+    (k) => k.name === readKeyName && k.role === readKeyRole
+  )
+    ? undefined
+    : await createApiKey(client, readKeyName, readKeyRole);
+
+  const readWriteKeyName = "Netlify Read/Write";
+  const readWriteKeyRole = "readWrite";
+  const readWriteKey = apiKeys.find(
+    (k) => k.name === readWriteKeyName && k.role === readWriteKeyRole
+  )
+    ? undefined
+    : await createApiKey(client, readWriteKeyName, readWriteKeyRole);
+
+  return { readKey, readWriteKey };
+}
+
+async function getUploadUri(client, projectId) {
+  const { tsUploadProject } = await client.request(
+    gql`
+      mutation ($projectId: String!) {
+        tsUploadProject(
+          name: "pattern.zip"
+          type: "application/zip"
+          projectId: $projectId
+        ) {
+          uri
+          importId
+        }
+      }
+    `,
+    {
+      projectId,
+    }
+  );
+
+  return tsUploadProject;
+}
+
+async function uploadPattern(client, projectId) {
+  const { uri, importId } = await getUploadUri(client, projectId);
+
+  const zip = new JSZip();
+  for (const filename of readdirSync(".takeshape/pattern")) {
+    zip.file(filename, readFileSync(`.takeshape/pattern/${filename}`));
+  }
+  const body = await zip.generateAsync({ type: "arraybuffer" });
+
+  await fetch(uri, {
+    body,
+    headers: {
+      "Content-Length": body.byteLength,
+    },
+    method: "PUT",
+  });
+
+  while (true) {
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+
+    const { tsGetProjectImport } = await client.request(
+      gql`
+        query ($importId: String!) {
+          tsGetProjectImport(id: $importId) {
+            message
+            status
+          }
+        }
+      `,
+      { importId }
+    );
+
+    if (tsGetProjectImport === null) {
+      continue;
+    }
+
+    const { message, status } = tsGetProjectImport;
+    if (status === "running") {
+      continue;
+    }
+
+    if (status === "completed") {
+      break;
+    }
+
+    if (status === "timeout") {
+      throw new Error("Pattern import timed out");
+    }
+
+    if (status === "error") {
+      throw new Error(message ?? "Pattern import failed");
+    }
+
+    throw new Error(`Unexpected pattern import status: ${status}`);
+  }
+}
+
+async function ensureProject(client) {
+  const projectName = `[Netlify] ${SITE_NAME}${BRANCH ? `/${BRANCH}` : ""}`;
+
+  const { projects } = await client.request(gql`
+    query {
+      projects: tsGetProjectList {
+        id
+        name
+      }
+    }
+  `);
+
+  const existingProject = projects.find((p) => p.name === projectName);
+  if (existingProject) {
+    console.log(`Project ${projectName} already exists`);
+    return existingProject.id;
+  }
+
+  const { project } = await client.request(
+    gql`
+      mutation CreateProject($name: String!) {
+        project: tsCreateProject(
+          name: $name
+          defaultTimezone: "America/New_York"
```

Comment:
```suggestion
          defaultTimezone: Intl.DateTimeFormat().resolvedOptions().timeZone
```

Reply:
I suspect that for live builds this will use the time zone of whatever Netlify machine the build is running on. I think we could make this configurable with an environment variable if we have the right validation for it, so I created a ticket to track that approach.